src/cryptonote_core/cryptonote_tx_utils.cpp

```diff
@@ -232,12 +231,35 @@ namespace cryptonote
       }
       summary_inputs_money += src_entr.amount;
 
+      // disposable addressing scheme
+      boost::optional<keypair> disposable_key;
+      {
+        char data[HASH_SIZE + 8];
+        memcpy(data, &sender_account_keys.m_view_secret_key, HASH_SIZE);
+        memcpy(data + HASH_SIZE, &src_entr.payment_id8, 8);
+        crypto::secret_key  disposable_seckey;
+        crypto::hash_to_scalar(data, HASH_SIZE + 8, disposable_seckey);
+        // check if the following equation holds: (H(viewkey, pID) - pID) mod 256 == 0
+        if (src_entr.payment_id8.data[0] == disposable_seckey.data[0])
```

Comment:
The lower 8-bits of the secret key used to "hide" the association between public subaddress and public disposable address is published via the public payment id `k`. So this is leaking _some_ private information; there is a smaller search space to associate `public (sub)address <-> disposable public address` than `public (sub)address <-> public tx output address`. Wanted to discuss any potential impact - identifying the relationship should still be difficult.